Fix overlay CSS

stevemk14ebr opened the pull request:
This should fix https://github.com/uiwjs/react-md-editor/issues/134 by removing the hardcoded padding and changing relative positioning to absolute. At least I hope it does, it does at least fix the overlay misalignment that was discovered when I tried to use the new work you mentioned in https://github.com/uiwjs/react-md-editor/issues/193.

jaywcjlove commented:
![image](https://user-images.githubusercontent.com/1680273/123567103-fefdc200-d7f3-11eb-9b7d-881238c06ed1.png)

The style change resulted in no alignment code.
@stevemk14ebr 

🔍 Inspect: https://vercel.com/wcj/react-md-editor/BFiKuTgY96pYGA26cP8xcpkP4DpP
✅ Preview: https://react-md-editor-git-fork-stevemk14ebr-master-wcj.vercel.app

stevemk14ebr commented:
I don't understand everything looks correct in the demo

jaywcjlove commented:
![image](https://user-images.githubusercontent.com/1680273/123583315-fddb8d80-d811-11eb-9093-ddb51c6465aa.png)

@stevemk14ebr Your style change is wrong.

stevemk14ebr commented:
Oh wow there's different behaviors between chrome and firefox here! I built this with firefox where it works, but chrome shows the issue as in your screenshot. Let me try to fix this

stevemk14ebr commented:
The latest commit fixes the text alignment but scroll doesn't work because the textarea gets clipped, I am not sure why this occurs?

stevemk14ebr commented:
should be fixed now! Scroll was broken since I changed the <pre> to position:absolute I changed that back to relative, and then fixed a padding issue on the parent div of the pre. Scroll and Text should align on both browsers now!

jaywcjlove commented:
![image](https://user-images.githubusercontent.com/1680273/123658115-02309680-d864-11eb-95b3-10a9fa36e2d8.png)

The gap is gone, what should I do? @stevemk14ebr

stevemk14ebr commented:
I've added the padding back, didn't realize you wanted that. The absolute position doesn't take into account the padding of the parent, so it must instead be inherited by the child (which I did). I also explictly set the word-wrap behaviors since it's more clear what's going on this way.

stevemk14ebr commented:
Looking at the diff it seems all I really did was change the margin on the pre!

jaywcjlove commented:
@stevemk14ebr thx!
